tools/harness-thci/OpenThread_WpanCtl.py

```diff
@@ -763,6 +725,21 @@ def __getCommissionerSessionId(self):
         print '%s call getCommissionerSessionId' % self.port
         return self.__sendCommand(WPANCTL_CMD + 'getprop -v Commissioner:SessionId')[0]
 
+    def __getJoinerState(self):
+        """ get joiner state """
+        maxDuration = 150  # seconds
+        t_end = time.time() + maxDuration
+        while time.time() < t_end:
+            joinerState = self.__stripValue(self.__sendCommand('sudo wpanctl getprop -v NCP:State')[0])
+            if joinerState == 'offline:commissioned':
+                return True
+            elif joinerState == 'offline':
```

Comment:
If join failed, the state is changed to CREDENTIALS_NEEDED
https://github.com/openthread/wpantund/blob/master/src/ncp-spinel/SpinelNCPInstance.cpp#L3831

according to src/wpantund/wpan-properties.h, it should be a different string, right?
#define kWPANTUNDStateCredentialsNeeded                         "associating:credentials-needed"